base_layer/core/src/validation/traits.rs

```diff
@@ -22,29 +22,29 @@
 
 use crate::validation::{and_then::AndThenValidator, error::ValidationError};
 
-pub type Validator<T, B> = Box<dyn Validation<T, B>>;
-pub type StatelessValidator<T> = Box<dyn StatelessValidation<T>>;
+pub type StatefulValidator<T, B> = Box<dyn StatefulValidation<T, B>>;
+pub type Validator<T> = Box<dyn Validation<T>>;
```

Comment:
I dont think we should rename `StatelessValidator` to just `validator`. I like renaming of `validator` to `StatefulValidator` thou.  Should we not rather stick to exactly clear calls of the names. Ei `StatelessValidator` and `StatefulValidator`

Reply:
Did you read my comment above? `StatelessValidator` is the "purest" form of a `Validator` - kind of like naming a rtait `AnimalessVehicle` :P